onnxruntime/core/session/inference_session.h

```diff
@@ -372,6 +372,11 @@ class InferenceSession {
     @return the name of the profile file.
     */
   std::string EndProfiling();
+  /**
+    * Return the stored start time of profiler
+    @return the stored start time of profiler
+    */
+  uint64_t GetProfilingStartTime();
```

Comment:
You can return a const-ref to session_profiler_ here so that this class doesn't become the middle man for returning other data members of the profiler. Then you can retrieve the attributes from session_profiler_ in SessionGetProfilingStartTime.

Reply:
Make sense. It can be more flexible. Just updated.